src/core/stdcpp/string.d

```diff
@@ -0,0 +1,204 @@
+    ref basic_string insert(size_type pos, const(T)* s, size_type n);
+    ref basic_string insert(size_type pos, size_type n, T c);
+    extern(D) ref basic_string insert(size_type pos, const(T)[] s)                      { insert(pos, s.ptr, s.length); return this; }
+
+    ref basic_string erase(size_type pos = 0, size_type len = npos);
+
+    // replace
+    // swap
+    void pop_back();
+
+    // String operations
+    size_type copy(T* s, size_type len, size_type pos = 0) const;
+
+    size_type find(ref const(basic_string) str, size_type pos = 0) const nothrow;
+    size_type find(const(T)* s, size_type pos = 0) const;
+    size_type find(const(T)* s, size_type pos, size_type n) const;
+    size_type find(T c, size_type pos = 0) const nothrow;
+
+    size_type rfind(ref const(basic_string) str, size_type pos = npos) const nothrow;
+    size_type rfind(const(T)* s, size_type pos = npos) const;
+    size_type rfind(const(T)* s, size_type pos, size_type n) const;
+    size_type rfind(T c, size_type pos = npos) const nothrow;
+
+    size_type find_first_of(ref const(basic_string) str, size_type pos = 0) const nothrow;
+    size_type find_first_of(const(T)* s, size_type pos = 0) const;
+    size_type find_first_of(const(T)* s, size_type pos, size_type n) const;
+    size_type find_first_of(T c, size_type pos = 0) const nothrow;
+
+    size_type find_last_of(ref const(basic_string) str, size_type pos = npos) const nothrow;
+    size_type find_last_of(const(T)* s, size_type pos = npos) const;
+    size_type find_last_of(const(T)* s, size_type pos, size_type n) const;
+    size_type find_last_of(T c, size_type pos = npos) const nothrow;
+
+    size_type find_first_not_of(ref const(basic_string) str, size_type pos = 0) const nothrow;
+    size_type find_first_not_of(const(T)* s, size_type pos = 0) const;
+    size_type find_first_not_of(const(T)* s, size_type pos, size_type n) const;
+    size_type find_first_not_of(T c, size_type pos = 0) const nothrow;
+
+    size_type find_last_not_of(ref const(basic_string) str, size_type pos = npos) const nothrow;
+    size_type find_last_not_of(const(T)* s, size_type pos = npos) const;
+    size_type find_last_not_of(const(T)* s, size_type pos, size_type n) const;
+    size_type find_last_not_of(T c, size_type pos = npos) const nothrow;
+
+    basic_string substr(size_type pos = 0, size_type len = npos) const;
+
+    int compare(ref const(basic_string) str) const nothrow;
+    int compare(size_type pos, size_type len, ref const(basic_string) str) const;
+    int compare(size_type pos, size_type len, ref const(basic_string) str, size_type subpos, size_type sublen) const;
+    int compare(const(T)* s) const;
+    int compare(size_type pos, size_type len, const(T)* s) const;
+    int compare(size_type pos, size_type len, const(T)* s, size_type n) const;
+
+    // D helpers
+    alias as_array this;
+    extern(D)        T[] as_array()         { return data()[0 .. size()]; }
```

Comment:
camelCase

Reply:
I'm not mixing naming standards within a single class!